### PR TITLE
Fix a ton of lint issues

### PR DESCRIPTION
**Pull request by jaseemabid**

### CODE REVIEW

File: src/io/fabric8/Fabric8Commands.groovy [hunk 1]
```diff
@@ -453,7 +453,7 @@ def getIssueComments(project, id, githubToken = null) {
     def apiUrl = new URL("https://api.github.com/repos/${project}/issues/${id}/comments")
     echo "getting comments for ${apiUrl}"
 
-    def HttpURLConnection connection = apiUrl.openConnection()
+    HttpURLConnection connection = apiUrl.openConnection()
```
Comment (sthaha) [inline src/io/fabric8/Fabric8Commands.groovy]: new to groovy ... could you please point to the rule it breaks?

File: src/io/fabric8/Fabric8Commands.groovy [hunk 2]
```diff
@@ -955,37 +944,36 @@ def openShiftImageStreamExists(String name){
...
-    return false;
+    return false
 }
 
 @NonCPS
 def openShiftImageStreamInstall(String name, String location){
     if (openShiftImageStreamExists(name)) {
         echo "ImageStream ${name} does not exist - installing ..."
         try {
-            def result = sh(returnStdout: true, script: 'oc create -f  ${location}')
```
Comment (sthaha) [inline src/io/fabric8/Fabric8Commands.groovy]: why is this deleted?

File: src/io/fabric8/Utils.groovy [hunk 3]
```diff
@@ -162,7 +161,7 @@ boolean isUseDockerSocket() {
     echo "WARNING: Failed to find the getUseDockerSocketFlag() flag on the PipelineConfiguration object - probably due to the jenkins plugin `kubernetes-pipeline-plugin` version: ${e}"
     e.printStackTrace()
   }
-  return supportsOpenShiftS2I() ? false : true;
+  return supportsOpenShiftS2I() ? false : true
```
Comment (sthaha) [inline src/io/fabric8/Utils.groovy]: how about `return !supportsOpenShiftS2I()`

File: vars/deployOpenShiftSnapshot.groovy [hunk 4]
```diff
@@ -38,10 +38,9 @@ def call(body) {
     }
     // cant use writeFile as we have long filename errors
     sh "echo '${yaml}' > snapshot.yml"
-    def template = false
-    if (yaml.contains('kind: Template')){
-        template = true
-    }
+
+    def template = yaml.contains('kind: Template')
```
Comment (sthaha) [inline vars/deployOpenShiftSnapshot.groovy]: good catch!

File: vars/getNewVersion.groovy [hunk 5]
```diff
@@ -9,14 +9,11 @@ def call(body) {
 
     // until we fix up pushing tags to remote repos with correct secrets lets default to use the short commit sha as the version
     echo 'NOTE: until we support pushing tags to remote repos with correct secrets lets default to use the short commit sha as the version'
-    def version 
     try {
-        version = sh(script: 'git rev-parse --short HEAD', returnStdout: true).toString().trim()
-    } catch (err){
+        def version = 'v' + sh(script: 'git rev-parse --short HEAD', returnStdout: true).toString().trim()
```
Comment (sthaha) [inline vars/getNewVersion.groovy]: [`error`][error] aborts the build in a graceful manner so, I think there nothing wrong with the old implementation which catches any error while executing git command and  gracefully terminates the build with a message.

[error]: https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#code-error-code-error-signal

File: vars/updatePropertyVersion.groovy [hunk 6]
```diff
@@ -44,10 +42,8 @@ def call(body) {
 
         sh "git push origin versionUpdate${uid}"
 
-        rs = flow.createPullRequest("[CD] Update release dependencies","${config.project}","versionUpdate${uid}")
-
+        return flow.createPullRequest("[CD] Update release dependencies","${config.project}","versionUpdate${uid}")
```
Comment (sthaha) [inline vars/updatePropertyVersion.groovy]: would returning from a closure exit the method? 
Also if you want to make the structure flat, you could use golang's idiom of returning early e.g. 

```groovy
if (!updated) {
  return
}
container(name: 'clients') {
 // ...
 return flow.createPullRequest(...)
}
```

but I am not sure if returning within a closure exits from the caller method (will test this soon) so if it only exists the closure, then this method may not return what `flow.createPull` returned.

File: src/io/fabric8/Fabric8Commands.groovy (hunk 1 above)
Comment (jaseemabid) [reply src/io/fabric8/Fabric8Commands.groovy]: @sthaha def is kind of untyped var definition, and if you define a variable with an explicit type, its not necessary. http://www.groovy-lang.org/semantics.html#_variable_definition

File: src/io/fabric8/Fabric8Commands.groovy (hunk 2 above)
Comment (jaseemabid) [reply src/io/fabric8/Fabric8Commands.groovy]: Oops. The linter warned that the variable is unused and removed it. It should be around. I caught a bunch of these when I checked the diff in the end, but this one slipped through.

File: src/io/fabric8/Utils.groovy (hunk 3 above)
Comment (jaseemabid) [reply src/io/fabric8/Utils.groovy]: @sthaha Yes, that works too. This is just automatic semi colon deletion, but I can do that too.

File: vars/deployOpenShiftSnapshot.groovy (hunk 4 above)
Comment (jaseemabid) [reply vars/deployOpenShiftSnapshot.groovy]: Yay!

File: vars/getNewVersion.groovy (hunk 5 above)
Comment (jaseemabid) [reply vars/getNewVersion.groovy]: @sthaha conceptually equivalent, but version might be undefined if catch returns early, I made the scope of the variable shorter and explicit inside the try block.

Comment (jaseemabid) [reply vars/getNewVersion.groovy]: @sthaha  I'm just reducing the scope of the version variable to the try block where it actually lives and removing a mutation.

**Comment by jaseemabid** [comment]
@sthaha I've pushed a new commit squashed to previous one. There is a much more saner commit message as well ;)

File: src/io/fabric8/Fabric8Commands.groovy [hunk 7]
```diff
@@ -71,7 +71,7 @@ def isArtifactAvailableInRepo(String repo, String groupId, String artifactId, St
         connection.connect()
         new InputStreamReader(connection.getInputStream(), "UTF-8")
         return true
-    } catch (FileNotFoundException e1) {
+    } catch (FileNotFoundException ignored) {
```
Comment (sthaha) [inline src/io/fabric8/Fabric8Commands.groovy]: IIUC, the `catch ` works without declaring `ignored` variable?

File: vars/deleteGitHubBranch.groovy [hunk 8]
```diff
@@ -29,7 +29,7 @@ def deleteBranch(String project, String branch, String authString){
       connection.connect()
 
       if (!isSuccessfulCode(connection.getResponseCode())) {
-        error ("DELETE was not successful: " + connection.getResponseCode() + " "+ connection.getResponseMessage());
+        error ("DELETE was not successful: " + connection.getResponseCode() + " "+ connection.getResponseMessage())
```
Comment (sthaha) [inline vars/deleteGitHubBranch.groovy]: minor: doesn't this look better? 
```groovy
  error "DELETE was not successful: ${connection.getResponseCode()} ${connection.getRespsonseMessage()}" 
```

File: src/io/fabric8/Fabric8Commands.groovy [hunk 9]
```diff
@@ -486,7 +486,7 @@ def waitUntilSuccessStatus(project, ref) {
 
     def apiUrl = new URL("https://api.github.com/repos/${project}/commits/${ref}/status")
     waitUntil {
-        def HttpURLConnection connection = apiUrl.openConnection()
+        HttpURLConnection connection = apiUrl.openConnection()
         if (githubToken != null && githubToken.length() > 0) {
```
Comment (sthaha) [inline src/io/fabric8/Fabric8Commands.groovy]: isn't the idiom `githubToken?.length() > 0 ` ?

File: vars/getNewVersion.groovy (hunk 5 above)
Comment (sthaha) [reply vars/getNewVersion.groovy]: ok .. I don't feel strongly about the new or the old. To me both are equally good. The old one had a return at the end of the method which (to me) was simpler

File: vars/updatePropertyVersion.groovy [hunk 10]
```diff
@@ -33,21 +33,15 @@ def call(body) {
...
         sh 'chmod 600 /root/.ssh-git/ssh-key.pub'
         sh 'chmod 700 /root/.ssh-git'
 
         sh "git push origin versionUpdate${uid}"
-
-        rs = flow.createPullRequest("[CD] Update release dependencies","${config.project}","versionUpdate${uid}")
-
       }
-      return rs
+      return flow.createPullRequest("[CD] Update release dependencies","${config.project}","versionUpdate${uid}")
```
Comment (sthaha) [inline vars/updatePropertyVersion.groovy]: should you be inside the `container('clients')` to be able to make a `PR`?

File: src/io/fabric8/Fabric8Commands.groovy (hunk 9 above)
Comment (jaseemabid) [reply src/io/fabric8/Fabric8Commands.groovy]: If the token itself can be null, you will need `githubToken && githubToken.length() > 0` at least. 

I'm sure we will find a lot more like this, no need to fix it all in one patch.

File: vars/updatePropertyVersion.groovy (hunk 10 above)
Comment (jaseemabid) [reply vars/updatePropertyVersion.groovy]: https://github.com/fabric8io-images/builder-clients (needs to be documented in the wiki :/) is just another container with a whole bunch of our binaries thrown in together. `createPullRequest` works by sending a HTTP post to github, so that should work anywhere as far as I understand now.

File: vars/mavenCanaryRelease.groovy [hunk 11]
```diff
@@ -36,12 +36,12 @@ def call(body) {
...
         }
 
-        bayesianScanner(body);
+        bayesianScanner(body)
     }
 
 
 
-    sonarQubeScanner(body);
+    sonarQubeScanner(body)
```
Comment (bartoszmajsak) [inline vars/mavenCanaryRelease.groovy]: Ever wondered why is this here and if your infra supports it? Seems to me like a thing no-one needs right now.
